app/Services/Three/SignupService.php

```diff
@@ -42,7 +42,9 @@ public function __construct(SignupRepository $signup, Blink $blink)
      */
     public function create($data, $transactionId)
     {
-        $signup = $this->signup->create($data);
+        $northstarId = isset($data['northstar_id']) ? $data['northstar_id'] : auth()->id();
```

Comment:
@DFurnes changed this here based on your last comment to decide which to send over!

Reply:
One catch - we want to make sure that only admins can set a custom user ID (since we wouldn't want a normal user to be able to hack someone else's reportbacks). I think it makes sense to handle this all in the controller, and have a `token()->role()` check there to see if we should use the provided `northstar_id` parameter.